salt/modules/file.py

```diff
@@ -1549,18 +1549,19 @@ def _regex_to_static(src, regex):
         return None
 
     try:
-        src = re.search(regex, src)
+        compiled = re.compile(regex, re.DOTALL)
+        src = [res.group() for res in map(compiled.search, src) if res]
     except Exception as ex:
         raise CommandExecutionError("{0}: '{1}'".format(_get_error_message(ex), regex))
 
-    return src and src.group() or regex
+    return src and src[0] or regex
```

Comment:
Why is this being done? I don't see this function being referenced in any of the changed code.